patches/minecraft/net/minecraft/item/crafting/RecipesBanners.java.patch

```diff
@@ -19,11 +19,11 @@
                          ItemStack itemstack2 = p_77572_1_.func_70301_a(j);
  
 -                        if (itemstack2.func_77973_b() == Items.field_151100_aR)
-+                        int color = getColor(itemstack2);
-+                        if (color != -1)
++                        final java.util.OptionalInt color = net.minecraftforge.oredict.DyeUtils.dyeDamageFromStack(itemstack2);
```

Comment:
@mezz Should the `DyeUtils#colorFromStack` method, which currently returns an `Optional<EnumDyeColor>`, also be changed to return just an `EnumDyeColor` with a sentinel (`null`) for a non-dye?